libeos-updater-util/avahi-service-file.h

```diff
@@ -44,4 +44,241 @@ gboolean eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
                                         GCancellable  *cancellable,
                                         GError       **error);
 
+/* new DNS-SD records format for ostree */
+/* TODO: gtkdocize it? */
+/* TXT records' values are basically serialized GVariants. Below all
+ * the keys and their GVariant types are described.
+ *
+ * The TXT records are served for the service type
+ * "_repo._ostree._tcp".
+ *
+ * Common fields for all versions of TXT records:
+ *
+ * - version, describes the version of the TXT records format:
+ *   - key: "v"
+ *   - type: "y" (byte)
+ *   - contents: a version number (note: it is not an ascii digit)
+ *
+ * Fields for version 1 of TXT records:
+ *
+ * - refs bloom filter, a Bloom filter that contains all the refs the host has
+ *   - key: "rb"
+ *   - type: "(yyay)" (tuple containing a byte, a byte and an array of bytes)
+ *   - contents: first byte is the "k" parameter, second byte is the
+ *               hash id, an array of bytes are the bloom filter bits
+ *
+ * - repository index, a number identifying an OSTree repository
+ *   - key: "ri"
+ *   - type: "q" (uint16)
+ *   - contents: it gets appended to the host's URL as "/%u"; 0 is a special
+ *               case - it is not appended, the host's URL serves as a home
+ *               of the repository
+ *
+ * - summary timestamp,
+ *   - key: "st",
+ *   - type: "t" (uint64)
+ *   - contents: a unix utc timestamp of the summary, ideally telling when
+ *               the original summary was created, otherwise it could also
+ *               be the modification time of the summary file on host
+ */
+
+/**
+ * EOS_OSTREE_AVAHI_OPTION_FORCE_VERSION_Y:
+ *
+ * Tells which version of DNS-SD records should be generated. Also,
+ * tells which set of options will be used when during the check or
+ * generation.
+ *
+ * Currently there is only one version available: 1.
+ *
+ * The options specific for the version 1 are:
+ *
+ * - %EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y
+ * - %EOS_OSTREE_AVAHI_OPTION_BLOOM_K_Y
+ * - %EOS_OSTREE_AVAHI_OPTION_BLOOM_SIZE_U
+ * - %EOS_OSTREE_AVAHI_OPTION_REPO_INDEX_Q
+ * - %EOS_OSTREE_AVAHI_OPTION_PORT_Q
+ * - %EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y
+ * - %EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_CUSTOM_SIZE_T
+ *
+ * Default value of this option (if not overridden) is 1.
+ */
+#define EOS_OSTREE_AVAHI_OPTION_FORCE_VERSION_Y "force-version"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y:
+ *
+ * Specifies the ID of the hashing function for the bloom filter.
+ *
+ * Currently there is only one hashing function available:
+ *
+ * - %EOS_OSTREE_AVAHI_STR_BLOOM_HASH_ID
+ *
+ * Default value of this option (if not overridden) is
+ * %EOS_OSTREE_AVAHI_STR_BLOOM_HASH_ID.
+ */
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y "bloom-hash-id"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_BLOOM_K_Y:
+ *
+ * Specifies the k parameter for the bloom filter. It roughly
+ * translates to how many times an element will be hashed before using
```

Comment:
I wrote roughly because I thought we do something more than just hashing the element `k` times. Fixed.